### PR TITLE
boards: esp32s3_devkitc: initial board support

### PR DESCRIPTION
**Pull request by wmrsouza**
Initial board files created to support the esp32s3_devkitc board.

### CODE REVIEW

File: boards/xtensa/esp32s3_devkitc/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+You can debug an application in the usual way. Here is an example for the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: esp32s3_devkitc
+   :goals: debug
+
+.. _`JTAG debugging for ESP32-S3`: https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-guides/jtag-debugging/
+.. _`OpenOCD`: https://github.com/openocd-org/openocd
+.. _`ESP32-S3 DevKitC`: https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html
```
Comment (sylvioalves) [inline boards/xtensa/esp32s3_devkitc/doc/index.rst:272]: Check duplicated reference. Move all to `References` section.

**Review by uLipe — DISMISSED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]
Needs to be migrated to hwmv2

**Comment by wmrsouza** [comment]
> Needs to be migrated to hwmv2

It was migrated tgo hwmv2

File: boards/espressif/esp32s3_devkitc/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+   to the board default configuration file.
+   ```
+   CONFIG_BOOTLOADER_MCUBOOT=y
+   ```
+
+Sysbuild
+========
+
+The sysbuild makes possible to build and flash all necessary images needed to
+bootstrap the board with the ESP32 SoC.
```
Comment (sylvioalves) [inline boards/espressif/esp32s3_devkitc/doc/index.rst:170]: Can you rename from `ESP32`to `ESP32-S3`?

File: boards/espressif/esp32s3_devkitc/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+To build the sample application using sysbuild use the command:
+
+.. zephyr-app-commands::
+   :tool: west
+   :app: samples/hello_world
+   :board: esp32s3_devkitc
```
Comment (sylvioalves) [inline boards/espressif/esp32s3_devkitc/doc/index.rst:177]: I think board name should be `esp32s3_devkitc/esp32s3/procpu`.

File: boards/espressif/esp32s3_devkitc/esp32s3_devkitc_esp32s3_procpu.dts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+&wdt0 {
+	status = "okay";
+};
+
+&trng0 {
+	status = "okay";
+};
+
+&usb_serial {
```
Comment (sylvioalves) [inline boards/espressif/esp32s3_devkitc/esp32s3_devkitc_esp32s3_procpu.dts:142]: Duplicated entry, please remove it.

File: boards/espressif/esp32s3_devkitc/doc/index.rst (hunk 3 above)
Comment (wmrsouza) [reply boards/espressif/esp32s3_devkitc/doc/index.rst]: done

File: boards/espressif/esp32s3_devkitc/doc/index.rst (hunk 2 above)
Comment (wmrsouza) [reply boards/espressif/esp32s3_devkitc/doc/index.rst]: done

File: boards/xtensa/esp32s3_devkitc/Kconfig.defconfig [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD
+	default "esp32s3_devkitc"
+	depends on BOARD_ESP32S3_DEVKITC
+
+config ENTROPY_GENERATOR
+	default y
+
+config HEAP_MEM_POOL_SIZE
```
[the remark this answers is not in the document]
Comment (wmrsouza) [reply boards/xtensa/esp32s3_devkitc/Kconfig.defconfig]: done

**Review by sylvioalves — APPROVED** [review]

**Comment by wmrsouza** [comment]
@nordicjm could you, please, review?

**Review by nordicjm — APPROVED** [review]

**Pull request merged**